### ISSUE TITLE
GatewayCluster with kubernetes backend fails to start after update to daskhub v4.5.4

### ISSUE THREAD

**Issue by bolliger32** [dask/dask-gateway#348]
**What happened**:
I received the following error: `GatewayClusterError: Cluster 'adrastea.b4286778ea9b49f4b4264f982f5b278d' failed to start, see logs for more information`. The logs suggest that the `dask-scheduler` command is missing an argument after `--host`, which looks intentional based on [this code](https://github.com/dask/dask-gateway/blob/e409f0e87f45e0a51fd7c009b5ec010bc5253bf1/dask-gateway-server/dask_gateway_server/backends/kubernetes/controller.py#L1029). Here are the logs:
![image](https://user-images.githubusercontent.com/4801430/98746312-1756c480-236a-11eb-8b41-1c09d6cf04f2.png)

This occurred upon update from v4.5.3 of the daskhub chart to v4.5.4. Note that several other issues occurred related to jupyterhub. I eventually worked my way through those and ultimately just deleted and recreated our GKE cluster. That fixed these other issues (primarily related to authentication) but this one remains.

**What you expected to happen**:
A working GatewayCluster object to be returned from the `gateway.new_cluster()` call

**Minimal Complete Verifiable Example**:
I'd imagine a lot of the reproducibility depends on our specific GKE infrastructure and chart config, but the actual code that raises this bug is just

```python
import dask_gateway

gateway = dask_gateway.GatewayCluster()
cluster = gateway.new_cluster()
```

**Anything else we need to know?**:

**Environment**:

- GKE cluster
- daskhub chart version: 4.5.4
- image: custom image running the following
  - Dask version: 2.30.0
  - Python version: 3.8.6
  - Operating System: ubuntu
  - Install method (conda, pip, source): conda

**Comment by jcrist** [comment]
Hmmm, that's odd. The command passed has an empty string, which should be forwarded correctly through k8s and still parse correctly (the pod spec takes a list of args, and `""` is a valid arg). We also have tests for this, and our k8s tests passed fine with these changes. This may be an esoteric bug, possible culprits:

- The shell used in your image differs from the shell in our images, and that matters for some reason?
- [Our default images use `tini` as an entrypoint](https://github.com/dask/dask-gateway/blob/c2583548df19359d24031e1dd9161c616d3bed50/dask-gateway/Dockerfile#L44), perhaps this matters for some reason?
- Perhaps a k8s difference/bug? I'm not sure what all touches the container `args` before things start up.

If you try things with the default images do they work fine? That would help figure out whether the images are the culprit or if it's a k8s difference.

Switching the `--host` arg to to `"0.0.0.0"` instead might work, but I'm curious why things pass with our images and fail with yours.

(*note - I'm off for the next week, so responses here may be a bit slow*)

**Comment by bolliger32** [comment]
Interesting. Thanks for the detailed set of possible causes! Using the default image for the scheduler doesn't seem to change anything, so i doubt that's the error. I will see if it matters if I switch to the default image for the notebook (it's not currently in our list of options so will take a sec to update the helm chart and add it).

The k8s version is 1.16.13 - not sure how that compares to whats being used in your tests.

To add to the confusion, I sometimes don't get that detailed output from the dask-scheduler command showing up in the logs. Most of the time it just returns relatively little info:
![image](https://user-images.githubusercontent.com/4801430/98748817-51769500-236f-11eb-8352-7921df6b690f.png)

**Comment by bolliger32** [comment]
@jcrist I take it back! I was not changing my cluster options correctly and hadn't actually switched images. When I use the default image it works! I'll dig into the couple of things you mentioned to figure out why our image is causing this behavior

**Comment by bolliger32** [comment]
In the end, I just decided to go with the default scheduler and use our customized image just on the worker. Seems to work if the important packages are pinned to the same version across these images. I'm sure it wouldn't take too much digging to figure out what was going on. Let me know if you think that would be helpful and I'm happy to give it a bit more effort. Otherwise, I think we can close this

**Comment by jcrist** [comment]
I'd really like to figure out what's going on here. Can you post the `Dockerfile` for your failing image (or an equivalent one that still fails)? That would be helpful in figuring out what the issue is.

**Comment by bolliger32** [comment]
[This](https://github.com/RhodiumGroup/docker_images/blob/3345859ca46f2045d4b9cd2c0d44ee6e8b915fab/worker/Dockerfile_scheduler) is what we were using for a Dockerfile. The conda env referenced in the file is [here](https://github.com/RhodiumGroup/docker_images/blob/3345859ca46f2045d4b9cd2c0d44ee6e8b915fab/shared_resources/scheduler_environment.yml).

Note that some commands in the [prepare.sh](https://github.com/RhodiumGroup/docker_images/blob/3345859ca46f2045d4b9cd2c0d44ee6e8b915fab/worker/prepare.sh) script run by tini fail when run on this scheduler image. This is an artifact of also using that startup script for our worker images, which are much heavier weight and were taking too long to spin up as the scheduler. This did not cause issues with the previous helm chart but could be part of the reason we were getting the issue with the dask-scheduler command.

**Comment by bolliger32** [comment]
@consideRatio hmm...it's been awhile but I can confirm that something between nov 2020 and now has solved this issue and we are now using our custom scheduler images (with daskhub 2021.8.1). I'm pretty certain you are right - we definitely were not explicitly disabling the `hub.networkPolicy` setting. We are doing that now (actually have implemented something similar to #352 in our `values.yml` file) and are able to use the custom image fine. I'm not sure if this was the underlying issue, but regardless - this is no longer a problem. I'd be in favor of closing the issue unless anyone would like to leave it open.

**Comment by consideRatio** [comment]
Thanks for following up @bolliger32!
